src/fides/api/ops/db/base_class.py

```diff
@@ -296,15 +294,8 @@ def save(self, db: Session) -> FidesBase:
         """Save the current object over an existing row in the database."""
         if hasattr(self, "key"):
             key = getattr(self, "key")
-
-            is_valid = False
-            if key is not None:
-                is_valid = len(ALLOWED_CHARS.sub("", key)) == 0
-
-            if not is_valid:
-                raise KeyValidationError(
-                    f"Key '{key}' on {self.__class__.__name__} is invalid."
-                )
```

Comment:
This seems like it has some unintentional side effects.  This validation was preventing "<" and ">" characters from getting saved as fides keys, and that's no longer the case.


 The Fideslang fides key validation allows "<" ">" characters for the purpose of being able to serialize the Saas Configs in memory which often have a placeholder key until the user gives it an actual key:

```yaml
dataset:
  - fides_key: <instance_fides_key>
    name: Amplitude
    description: A sample dataset representing the Amplitude connector for Fides
    collections:
   ...
``` 

But it still seems useful to prevent those characters when we actually save the resource.

Reply:
I see, so you're saying it makes more sense to have a _stricter_ character check here because it could prevent a bug where a user doesn't assign a proper value and a placeholder key gets saved?

Reply:
Yes, this wasn't intentional but it seems useful in hindsight.